[jevan13] iP

jevan13 opened the pull request:
# Yapper
> "If it's flipping hamburgers at McDonald's, be the best hamburger flipper in the world. Whatever it is you do you have to master your craft." -Snoop Dogg

Yapper frees your mind of having to remember things you need to do. It's,

text-based
easy to learn
~~FAST~~ **SUPER FAST** to use
All you need to do is,

download it from [here](https://github.com/jevan13/ip/releases/tag/A-Jar).
1. double-click it.
2. add your tasks.
3. let it manage your tasks for you 😉
**And it is FREE!**

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

Commit:
Rename, Greet, Exit

Commit:
Echo

Commit:
Add, List

Commit:
Mark as Done

Commit:
ToDos, Events, Deadlines

Commit:
Handle Errors

Commit:
Delete

Commit:
A-TextUiTesting

Commit:
Level-5 Handle Errors: Changed to Exceptions

Chatbot will no long terminate upon wrong command

Commit:
Revert "Level-5 Handle Errors: Changed to Exceptions"

This reverts commit 61e79b90bd049ac4087b77e4cf4e2cf9b50bdd70.

Commit:
Revert "Revert "Level-5 Handle Errors: Changed to Exceptions""

This reverts commit 13c53f85913b2f8bcfe076c34d62c8392db1065f.

Commit:
Save

Commit:
Dates and Times

Commit:
A-MoreOOP

jevan13 committed:
Add all classes to Yapper package

jevan13 committed:
Merge branch 'branch-A-Packages'

* branch-A-Packages:
  Add all classes to Yapper package

jevan13 committed:
Add Time

jevan13 committed:
Merge branch 'add-gradle-support'

* add-gradle-support:
  build.gradle: Prevent generating a second JAR file
  Bump gradle and lib version
  Add Gradle support

jevan13 committed:
Add Gradle

jevan13 committed:
Merge branch 'A-Gradle'

* A-Gradle:
  Add Gradle

jevan13 committed:
Add ToDoTest and ParserTest

jevan13 committed:
Add Tests

jevan13 committed:
Merge branch 'A-JUnit'

* A-JUnit:
  Add Tests
  Add ToDoTest and ParserTest

jevan13 committed:
Add JavaDoc

jevan13 committed:
Merge branch 'A-JavaDoc'

* A-JavaDoc:
  Add JavaDoc

jevan13 committed:
Add Coding Standard

jevan13 committed:
Add find

jevan13 committed:
Merge tag 'A-CodingStandard'

* tag 'A-CodingStandard':
  Add Coding Standard

jevan13 committed:
Merge conflict

gnoossk commented on src/main/java/Yapper/TaskList.java:
```diff
@@ -0,0 +1 @@
    public void exit() {
        System.out.println("Yapper shall yap next time!");
    }
}
```
I like the usage of "to___" for String methods

gnoossk commented on src/main/java/Yapper/Task.java:
```diff
@@ -0,0 +1 @@
    public boolean contains(String word) {
        return this.name.contains(word);
    }
}
```
Not sure I like the getter method name for isDone. Can be specified to something like `hasDone` rather than `getDone`

thony-ui commented on src/main/java/Yapper/Storage.java:
for boolean values such as 
```java
done
```

it should be named as 

```java
isDone
```
for clarity

thony-ui commented on src/main/java/Yapper/Task.java:
I think the getter method should be

```java
getIsDone
```

because the field that you specified is ```isDone```

thony-ui commented on src/main/java/Yapper/TaskList.java:
This is where the use of ```Enumeration``` may be useful. Notice how you check for equality in the "if" statements? You can abstract this in the ```Enumeration``` java class and utilise that.

thony-ui commented on src/test/java/Yapper/TaskListTest.java:
Good use of test cases. However, I would also test for edge cases if it doesn't match the format.

jevan13 committed:
Add CheckStyle, and change Line Separator settings

jevan13 committed:
Merge tag 'A-CheckStyle'

* tag 'A-CheckStyle':
  Add CheckStyle, and change Line Separator settings

jevan13 committed:
Add Gui, Fix CheckStyle

jevan13 committed:
Merge tag 'Level-10'

* tag 'Level-10':
  Add Gui, Fix CheckStyle

jevan13 committed:
Reformat src

jevan13 committed:
Change Line Separator to \r

jevan13 committed:
Change Line Separator to \n

jevan13 committed:
Change Line Separator to \r\n

jevan13 committed:
Change Line Separator to \n using mac2unix

jevan13 committed:
Change Line Separator to \n using mac2unix for Test

jevan13 committed:
Add Assertions

There are several methods that require an assumption of parameter values.

Adding Assertions will define these assumptions so that the runtime can verify them.
If the runtime detects an assertion failure, it typically takes some drastic action such as terminating the execution with an error message.

jevan13 committed:
Improve Code Quality

Critically examines the code and refactor to improve the code quality where necessary.
Closely followed the 'Code Quality' topics learnt so far, rather than intuition about code quality.

jevan13 committed:
Merge pull request #2 from jevan13/A-Assertions

A-Assertions

jevan13 committed:
Merge pull request #3 from jevan13/A-CodeQuality

A-CodeQuality

jevan13 committed:
Add B-Snooze

Users may want to change the start or end time of a task, like a deadline or event.

jevan13 committed:
Fix setIsDone method

jevan13 committed:
Merge branch 'BCD-Extension'

* BCD-Extension:
  Fix setIsDone method

# Conflicts:
#	src/main/java/yapper/Task.java

jevan13 committed:
Add a representative screenshot of the product

jevan13 committed:
Rename Screenshot 2024-09-18 at 12.55.21 AM.png to Ui.png

jevan13 committed:
Update README.md

jevan13 committed:
Add some changes to task toString() methods.

jevan13 committed:
Merge branch 'master' of https://github.com/jevan13/ip

* 'master' of https://github.com/jevan13/ip:
  Update README.md
  Rename Screenshot 2024-09-18 at 12.55.21 AM.png to Ui.png
  Add a representative screenshot of the product

jevan13 committed:
Update JUnit Tests

jevan13 committed:
Update JUnit Test

Add 2 more JUnit Test for Event toString() and toFile().
Required at least 2 methods unit tested well using JUnit.

jevan13 committed:
Move classes from Gui Package to yapper package.

Attempt to fix "Could not find or load main class Launcher" issue when running Yapper.jar

jevan13 committed:
Update mainClass in build.gradle

Changed mainClass to yapper.Launcher.
Fix previous error : Could not find or load main class Launcher.

jevan13 committed:
Add Welcome message

On launch, Yapper would send a message: "Hello. My name is Yapper. What can I do for you?".
Prompt user to enter input after launching the application.

jevan13 committed:
Add JavaDocs

Checkstyle required FXML to have JavaDocs.
Add info : Initialize the Main Window.
Under MainWindow.Initialize()

jevan13 committed:
Add terminating function

User input "bye" will exit by terminating the program and closing the window.

jevan13 committed:
Add assertions to postpone method for better validation and error checking

- Added assertions to check command length and number of inputs are valid
- Ensured task type and valid number of inputs are asserted
- Helps catch edge cases during development and improves code robustness

jevan13 committed:
Refactor findTask method for improved validation and handling

- Added check to trim and validate command input to handle edge cases
- Ensured wordToFind is extracted and trimmed before processing
- Improved error handling for empty search string
- Enhanced readability and consistency in validation logic

jevan13 committed:
Update Javadoc for task history loading method

- Expanded documentation to include information on file creation when no file exists
- Clarified exception handling by specifying YapperException for file read errors
- Improved overall clarity and completeness of method documentation

jevan13 committed:
Update Javadoc for Storage loading method

- Removed @throws YapperException as the method no longer throws exceptions
- Clarified the description of the method's functionality
- Improved overall documentation accuracy

jevan13 committed:
Refactor returnList method to improve readability and efficiency

- Added early return for empty task list with a clear message
- Simplified string concatenation by eliminating unnecessary variables
- Improved overall structure for better maintainability
